pkg/v21/resource/configmapmigration/create.go

```diff
@@ -66,8 +65,8 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	if cc.Client.TenantCluster.G8s == nil {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant clients not available")
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
-		reconciliationcanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
This doesn't work. We can't cancel the reconciliation because in new clusters we need to create app CRs.

Instead https://github.com/giantswarm/cluster-operator/pull/740 implements a defensive check. So if an app has user settings we don't create the app CR until the settings have been migrated.